### ISSUE TITLE
Version 5.0.0 seems like not compatible with Spring-Data-Jpa after maven packaged

### ISSUE THREAD

**Issue by dengbingaliyun** [joinfaces/joinfaces#1625]
OpenJdk 17.0.2

Joinfaces 5.0.0
Spring boot 3.0.0
Spring Data Jpa
Primefaces 12
Undertow

Spring Tool Suit 4

Under the development envirment is ok.
Then I used "maven clean package" to generate a jar file.

But more than 20,000 lines of Exception appeard in the log file after  I executed " java -jar omega-template-springboot.jar"

my pom.xml and application.log below:

pom.xml
`<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<groupId>org.myspring.springboot</groupId> 
	<artifactId>omega-template-springboot</artifactId>
	<version>2.0.0</version>

	 
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>3.0.0</version>
		<relativePath/> 
	</parent>

	<properties>
		<java.version>17</java.version>
		<joinfaces.version>5.0.0</joinfaces.version>
		<maven-model.version>3.6.0</maven-model.version>
		<poi-verion>4.0.0</poi-verion>
		<maven.build.timestamp.format>yyyy-MM-dd HH:mm:ss</maven.build.timestamp.format>
		
		
        <open-pdf.version>1.3.24</open-pdf.version>
        <poi.version>5.0.0</poi.version>
	</properties>
	
	
  	<dependencyManagement>
		<dependencies>
			<dependency>
				<groupId>org.joinfaces</groupId>
				<artifactId>joinfaces-dependencies</artifactId>
				<version>${joinfaces.version}</version>
				<type>pom</type>
				<scope>import</scope>
			</dependency>
		</dependencies>
	</dependencyManagement>

	<dependencies>
	
		<dependency>
			<groupId>org.joinfaces</groupId>
			<artifactId>jsf-spring-boot-starter</artifactId>
			<exclusions>
				<exclusion>
		            <groupId>org.springframework.boot</groupId>
		            <artifactId>spring-boot-starter-tomcat</artifactId>
		        </exclusion>

				<exclusion>
					<groupId>org.joinfaces</groupId>
					<artifactId>
						mojarra3-spring-boot-starter
					</artifactId>
				</exclusion>
			</exclusions>
		</dependency>
		
		
		
		
		
		
		
		<dependency>
		    <groupId>org.springframework.boot</groupId>
		    <artifactId>spring-boot-starter-undertow</artifactId>
		</dependency>
 		<dependency>
		    <groupId>org.springframework.boot</groupId>
		    <artifactId>spring-boot-starter-validation</artifactId>
		</dependency>
		
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-data-jpa</artifactId>
		</dependency>
		
		 
		
		

		
		<dependency>
			<groupId>org.joinfaces</groupId>
			<artifactId>primefaces-spring-boot-starter</artifactId> 
		</dependency>

		
		

		<dependency>
			<groupId>org.joinfaces</groupId>
			<artifactId>omnifaces-spring-boot-starter</artifactId>
		</dependency>
	
		
		
		<dependency>
			<groupId>org.joinfaces</groupId>
			<artifactId>mojarra4-spring-boot-starter</artifactId> 
		</dependency>
		
		<dependency>
			<groupId>org.jboss</groupId>
			<artifactId>jandex</artifactId>
			<version>2.4.3.Final</version>
		</dependency>
	
		
	   <dependency>
	        <groupId>commons-beanutils</groupId>
		    <artifactId>commons-beanutils</artifactId>
		    <version>1.9.3</version>
	   </dependency>

		
		 <dependency>
            <groupId>com.google.code.gson</groupId>
            <artifactId>gson</artifactId>
            <scope>runtime</scope>
        </dependency>
		<dependency>
			<groupId>org.apache.httpcomponents</groupId>
			<artifactId>httpclient</artifactId>
		</dependency>
		
		
        
		
        
        <dependency>
            <groupId>org.apache.poi</groupId>
            <artifactId>poi</artifactId>
            <version>${poi.version}</version>
        </dependency>
        <dependency>
            <groupId>org.apache.poi</groupId>
            <artifactId>poi-ooxml</artifactId>
            <version>${poi.version}</version>
            <exclusions>
                <exclusion>
                    <groupId>org.apache.xmlgraphics</groupId>
                    <artifactId>batik-all</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>de.rototor.pdfbox</groupId>
                    <artifactId>graphics2d</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.apache.santuario</groupId>
                    <artifactId>xmlsec</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.bouncycastle</groupId>
                    <artifactId>bcpkix-jdk15on</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>com.github.virtuald</groupId>
                    <artifactId>curvesapi</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.bouncycastle</groupId>
                    <artifactId>bcprov-jdk15on</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>com.github.librepdf</groupId>
            <artifactId>openpdf</artifactId>
            <version>${open-pdf.version}</version>
        </dependency>

		<dependency>
			<groupId>org.projectlombok</groupId>
			<artifactId>lombok</artifactId>
			<scope>provided</scope>
		</dependency>
		
		<dependency>
			<groupId>org.postgresql</groupId>
			<artifactId>postgresql</artifactId> 
		</dependency>
		
		

		
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-test</artifactId>
            <scope>test</scope>
        </dependency>
        

		<dependency>
			<groupId>org.primefaces.theme</groupId>
			<artifactId>omega</artifactId>
			<version>5.0.0</version>
			<scope>system</scope>
			<systemPath>${basedir}/src/main/resources/META-INF/resources/WEB-INF/lib/omega-theme-5.0.0-jakarta.jar</systemPath>
		</dependency>

	</dependencies>
	
	
	<profiles>
		<profile>
        
            <id>dev</id>
            <properties>
                <profiles.active>dev</profiles.active>
            </properties>
            <activation>
            	<activeByDefault>true</activeByDefault>
            </activation>
        </profile>
        <profile>
            
            <id>test</id>
            <properties>
                <profiles.active>test</profiles.active>
            </properties>
        </profile>
        <profile>
          
            <id>prod</id>
            <properties>
                <profiles.active>prod</profiles.active>
            </properties>
        </profile>
    </profiles>

	<build>
		<finalName>${project.artifactId}</finalName>
        <pluginManagement>
            <plugins>
                <plugin>
                    <groupId>org.joinfaces</groupId>
                    <artifactId>joinfaces-maven-plugin</artifactId>
                    <version>${joinfaces.version}</version>
                    <executions>
                        <execution>
                            <goals>
                                <goal>classpath-scan</goal>
                            </goals>
                        </execution>
                    </executions>
                </plugin>
                
                <plugin>  
	                <groupId>org.apache.maven.plugins</groupId>  
	                <artifactId>maven-archetype-plugin</artifactId>  
	                <version>3.0.0</version>  
	            </plugin>
            </plugins>
        </pluginManagement>

       <plugins>
       
       <plugin>
	        <groupId>org.joinfaces</groupId>
	        <artifactId>joinfaces-maven-plugin</artifactId>
	    </plugin>
       
       <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-failsafe-plugin</artifactId>
            </plugin>
        <plugin>
				<groupId>org.springframework.boot</groupId>
				<artifactId>spring-boot-maven-plugin</artifactId>
				<configuration>
					<mainClass>org.myspring.MainApplication</mainClass>
					<includeSystemScope>true</includeSystemScope>
				</configuration>
				<executions>
					<execution>
						<goals>
							<goal>repackage</goal>
							<goal>build-info</goal>
						</goals>
						<configuration>
                            <additionalProperties>
                                <encoding.source>UTF-8</encoding.source>
                                <encoding.reporting>UTF-8</encoding.reporting>
                                <java.source>${maven.compiler.source}</java.source>
                                <java.target>${maven.compiler.target}</java.target>
                            </additionalProperties>
                        </configuration>
					</execution>
				</executions>
			</plugin>
			
			<plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <configuration>
                    <encoding>UTF-8</encoding>
                    <source>17</source>
                    <target>17</target>
                </configuration>
            </plugin>
            <plugin>
            	<groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-resources-plugin</artifactId>
                <version>3.1.0</version>
                <configuration>
                    <encoding>UTF-8</encoding>
                </configuration>
            </plugin>
        
			
		</plugins>
		
		<resources>
			<resource>
	         <directory>src/main/resources</directory>
	            <filtering>true</filtering>
	            <excludes>
					<exclude>**/*.jar</exclude>
					<exclude>**/config/*.properties</exclude>
				</excludes>
        	</resource>
        	<resource>
	         <directory>src/main/resources/config</directory>
	            <filtering>true</filtering>
	            <includes>
	            	<include>application.properties</include>
	            	<include>application-${profiles.active}.properties</include>
	            </includes>
        	</resource>
		</resources>
	</build>

</project>`

application.log

```
2022-12-13T16:04:24.704+08:00  INFO 127368 --- [main] org.myspring.MainApplication             : Starting MainApplication v2.0.0 using Java 17.0.2 with PID 127368 (E:\workspaceSts4\omega-template-springboot\target\omega-template-springboot.jar started by dengb in E:\workspaceSts4\omega-template-springboot\target)
2022-12-13T16:04:24.709+08:00  INFO 127368 --- [main] org.myspring.MainApplication             : No active profile set, falling back to 1 default profile: "default"
2022-12-13T16:04:25.468+08:00  INFO 127368 --- [main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2022-12-13T16:04:25.495+08:00  INFO 127368 --- [main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 17 ms. Found 0 JPA repository interfaces.
2022-12-13T16:04:25.712+08:00  WARN 127368 --- [main] ocalVariableTableParameterNameDiscoverer : Using deprecated '-debug' fallback for parameter name resolution. Compile the affected code with '-parameters' instead or avoid its introspection: org.joinfaces.autoconfigure.scopemapping.CdiScopeAnnotationsAutoConfiguration
2022-12-13T16:04:25.717+08:00  WARN 127368 --- [main] ocalVariableTableParameterNameDiscoverer : Using deprecated '-debug' fallback for parameter name resolution. Compile the affected code with '-parameters' instead or avoid its introspection: org.joinfaces.autoconfigure.scopemapping.FacesScopeAnnotationsAutoConfiguration
2022-12-13T16:04:25.857+08:00  WARN 127368 --- [main] ocalVariableTableParameterNameDiscoverer : Using deprecated '-debug' fallback for parameter name resolution. Compile the affected code with '-parameters' instead or avoid its introspection: org.joinfaces.autoconfigure.faces.JakartaFaces3AutoConfiguration
2022-12-13T16:04:25.965+08:00  WARN 127368 --- [main] ocalVariableTableParameterNameDiscoverer : Using deprecated '-debug' fallback for parameter name resolution. Compile the affected code with '-parameters' instead or avoid its introspection: org.joinfaces.autoconfigure.javaxfaces.JsfBeansAutoConfiguration$Jsf2_3AutoConfiguration
2022-12-13T16:04:26.067+08:00  WARN 127368 --- [main] ocalVariableTableParameterNameDiscoverer : Using deprecated '-debug' fallback for parameter name resolution. Compile the affected code with '-parameters' instead or avoid its introspection: org.joinfaces.autoconfigure.undertow.UndertowAutoConfiguration
2022-12-13T16:04:26.272+08:00  INFO 127368 --- [main] o.j.a.u.UndertowAutoConfiguration        : Setting Undertow classLoader to META-INF/resources directory
2022-12-13T16:04:26.304+08:00  WARN 127368 --- [main] io.undertow.websockets.jsr               : UT026010: Buffer pool was not set on WebSocketDeploymentInfo, the default pool will be used
2022-12-13T16:04:26.325+08:00  INFO 127368 --- [main] io.undertow.servlet                      : Initializing Spring embedded WebApplicationContext
2022-12-13T16:04:26.328+08:00  INFO 127368 --- [main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 1567 ms
2022-12-13T16:04:26.332+08:00  WARN 127368 --- [main] ocalVariableTableParameterNameDiscoverer : Using deprecated '-debug' fallback for parameter name resolution. Compile the affected code with '-parameters' instead or avoid its introspection: org.joinfaces.autoconfigure.faces.FacesServletAutoConfiguration
2022-12-13T16:04:26.358+08:00  WARN 127368 --- [main] ocalVariableTableParameterNameDiscoverer : Using deprecated '-debug' fallback for parameter name resolution. Compile the affected code with '-parameters' instead or avoid its introspection: org.joinfaces.autoconfigure.primefaces.PrimefacesFileUploadServletContextAutoConfiguration
2022-12-13T16:04:26.371+08:00  WARN 127368 --- [main] ocalVariableTableParameterNameDiscoverer : Using deprecated '-debug' fallback for parameter name resolution. Compile the affected code with '-parameters' instead or avoid its introspection: org.joinfaces.autoconfigure.servlet.initparams.ServletContextInitParameterPropertiesAutoConfiguration
2022-12-13T16:04:26.465+08:00  INFO 127368 --- [main] org.omnifaces.ApplicationInitializer     : Using OmniFaces version 4.0.1
2022-12-13T16:04:27.134+08:00  INFO 127368 --- [main] org.joinfaces.ClasspathScanUtil          : Loading prepared scan result took 574.9531ms
2022-12-13T16:04:27.155+08:00  INFO 127368 --- [main] o.j.weld.environment.servletWeldServlet  : WELD-ENV-001008: Initialize Weld using ServletContainerInitializer
2022-12-13T16:04:27.178+08:00  INFO 127368 --- [main] org.jboss.weld.Version                   : WELD-000900: 5.1.0 (Final)
2022-12-13T16:04:27.293+08:00  INFO 127368 --- [main] o.s.d.j.r.cdi.JpaRepositoryExtension     : Activating CDI extension for Spring Data JPA repositories
2022-12-13T16:04:27.316+08:00  INFO 127368 --- [main] org.jboss.weld.Bootstrap                 : WELD-ENV-000020: Using jandex for bean discovery
2022-12-13T16:04:27.392+08:00  WARN 127368 --- [main] org.jboss.weld.Bootstrap                 : WELD-ENV-000010: Could not open the stream on the url jar:file:/E:/workspaceSts4/omega-template-springboot/target/omega-template-springboot.jar!/BOOT-INF/lib/joinfaces-autoconfigure-5.0.0.jarorg/joinfaces/autoconfigure/myfaces/MyFacesProperties$Annotation.class when adding to the jandex index.

java.io.FileNotFoundException: JAR entry BOOT-INF/lib/joinfaces-autoconfigure-5.0.0.jarorg/joinfaces/autoconfigure/myfaces/MyFacesProperties$Annotation.class not found in E:\workspaceSts4\omega-template-springboot\target\omega-template-springboot.jar
	at org.springframework.boot.loader.jar.JarURLConnection.throwFileNotFound(JarURLConnection.java:175)
	at org.springframework.boot.loader.jar.JarURLConnection.connect(JarURLConnection.java:98)
	at org.springframework.boot.loader.jar.JarURLConnection.getInputStream(JarURLConnection.java:162)
	at java.base/java.net.URL.openStream(URL.java:1161)
	at org.jboss.weld.environment.deployment.discovery.jandex.JandexFileSystemBeanArchiveHandler.addToIndex(JandexFileSystemBeanArchiveHandler.java:62)
	at org.jboss.weld.environment.deployment.discovery.jandex.JandexFileSystemBeanArchiveHandler.add(JandexFileSystemBeanArchiveHandler.java:55)
	at org.jboss.weld.environment.deployment.discovery.FileSystemBeanArchiveHandler.handleNestedFile(FileSystemBeanArchiveHandler.java:140)
	at org.jboss.weld.environment.deployment.discovery.FileSystemBeanArchiveHandler.handle(FileSystemBeanArchiveHandler.java:74)
	at org.jboss.weld.environment.deployment.discovery.jandex.JandexFileSystemBeanArchiveHandler.handle(JandexFileSystemBeanArchiveHandler.java:43)
	at org.jboss.weld.environment.deployment.discovery.AbstractDiscoveryStrategy.performDiscovery(AbstractDiscoveryStrategy.java:130)
	at org.jboss.weld.environment.servlet.WeldServletLifecycle.createDeployment(WeldServletLifecycle.java:298)
	at org.jboss.weld.environment.servlet.WeldServletLifecycle.initialize(WeldServletLifecycle.java:150)
	at org.jboss.weld.environment.servlet.EnhancedListener.onStartup(EnhancedListener.java:62)
	at org.joinfaces.servlet.ServletContainerInitializerRegistrationBean.lambda$customize$0(ServletContainerInitializerRegistrationBean.java:65)
	at org.springframework.boot.web.embedded.undertow.UndertowServletWebServerFactory$Initializer.onStartup(UndertowServletWebServerFactory.java:507)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:204)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:187)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at io.undertow.servlet.core.DeploymentManagerImpl.deploy(DeploymentManagerImpl.java:255)
	at org.springframework.boot.web.embedded.undertow.UndertowServletWebServerFactory.createManager(UndertowServletWebServerFactory.java:329)
	at org.springframework.boot.web.embedded.undertow.UndertowServletWebServerFactory.getWebServer(UndertowServletWebServerFactory.java:298)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.createWebServer(ServletWebServerApplicationContext.java:183)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:161)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:578)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:730)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:432)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:308)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1302)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1291)
	at org.myspring.MainApplication.main(MainApplication.java:14)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:95)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65)

2022-12-13T16:04:27.410+08:00  WARN 127368 --- [main] org.jboss.weld.Bootstrap                 : WELD-ENV-000010: Could not open the stream on the url jar:file:/E:/workspaceSts4/omega-template-springboot/target/omega-template-springboot.jar!/BOOT-INF/lib/joinfaces-autoconfigure-5.0.0.jarorg/joinfaces/autoconfigure/myfaces/MyFacesAutoConfiguration.class when adding to the jandex index.

java.io.FileNotFoundException: JAR entry BOOT-INF/lib/joinfaces-autoconfigure-5.0.0.jarorg/joinfaces/autoconfigure/myfaces/MyFacesAutoConfiguration.class not found in E:\workspaceSts4\omega-template-springboot\target\omega-template-springboot.jar
	at org.springframework.boot.loader.jar.JarURLConnection.throwFileNotFound(JarURLConnection.java:175)
	at org.springframework.boot.loader.jar.JarURLConnection.connect(JarURLConnection.java:98)
	at org.springframework.boot.loader.jar.JarURLConnection.getInputStream(JarURLConnection.java:162)
	at java.base/java.net.URL.openStream(URL.java:1161)
	at org.jboss.weld.environment.deployment.discovery.jandex.JandexFileSystemBeanArchiveHandler.addToIndex(JandexFileSystemBeanArchiveHandler.java:62)
	at org.jboss.weld.environment.deployment.discovery.jandex.JandexFileSystemBeanArchiveHandler.add(JandexFileSystemBeanArchiveHandler.java:55)
	at org.jboss.weld.environment.deployment.discovery.FileSystemBeanArchiveHandler.handleNestedFile(FileSystemBeanArchiveHandler.java:140)
	at org.jboss.weld.environment.deployment.discovery.FileSystemBeanArchiveHandler.handle(FileSystemBeanArchiveHandler.java:74)
	at org.jboss.weld.environment.deployment.discovery.jandex.JandexFileSystemBeanArchiveHandler.handle(JandexFileSystemBeanArchiveHandler.java:43)
	at org.jboss.weld.environment.deployment.discovery.AbstractDiscoveryStrategy.performDiscovery(AbstractDiscoveryStrategy.java:130)
	at org.jboss.weld.environment.servlet.WeldServletLifecycle.createDeployment(WeldServletLifecycle.java:298)
	at org.jboss.weld.environment.servlet.WeldServletLifecycle.initialize(WeldServletLifecycle.java:150)
	at org.jboss.weld.environment.servlet.EnhancedListener.onStartup(EnhancedListener.java:62)
	at org.joinfaces.servlet.ServletContainerInitializerRegistrationBean.lambda$customize$0(ServletContainerInitializerRegistrationBean.java:65)
	at org.springframework.boot.web.embedded.undertow.UndertowServletWebServerFactory$Initializer.onStartup(UndertowServletWebServerFactory.java:507)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:204)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:187)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at io.undertow.servlet.core.DeploymentManagerImpl.deploy(DeploymentManagerImpl.java:255)
	at org.springframework.boot.web.embedded.undertow.UndertowServletWebServerFactory.createManager(UndertowServletWebServerFactory.java:329)
	at org.springframework.boot.web.embedded.undertow.UndertowServletWebServerFactory.getWebServer(UndertowServletWebServerFactory.java:298)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.createWebServer(ServletWebServerApplicationContext.java:183)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:161)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:578)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:730)



.......



java.lang.ClassNotFoundException: jakarta.faces.bean.ManagedBean
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:587)
	at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:149)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at org.joinfaces.ClasspathScanUtil.lambda$readClassMap$0(ClasspathScanUtil.java:100)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at org.joinfaces.ClasspathScanUtil.readClassMap(ClasspathScanUtil.java:93)
	at org.joinfaces.ClasspathScanUtil.readClasses(ClasspathScanUtil.java:75)
	at org.joinfaces.ClasspathScanUtil.readClassMap(ClasspathScanUtil.java:59)
	at org.joinfaces.autoconfigure.FacesAnnotationProviderUtil.findPreparedScanResult(FacesAnnotationProviderUtil.java:85)
	at org.joinfaces.autoconfigure.mojarra.JoinFacesAnnotationProvider.getAnnotatedClasses(JoinFacesAnnotationProvider.java:52)
	at com.sun.faces.config.manager.tasks.FindAnnotatedConfigClasses.call(FindAnnotatedConfigClasses.java:89)
	at com.sun.faces.config.manager.tasks.FindAnnotatedConfigClasses.call(FindAnnotatedConfigClasses.java:40)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.sun.faces.config.ConfigManager.findAnnotations(ConfigManager.java:357)
	at com.sun.faces.config.ConfigManager.initialize(ConfigManager.java:250)
	at com.sun.faces.config.ConfigureListener.contextInitialized(ConfigureListener.java:179)
	at io.undertow.servlet.core.ApplicationListeners.contextInitialized(ApplicationListeners.java:187)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:219)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:187)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at io.undertow.servlet.core.DeploymentManagerImpl.deploy(DeploymentManagerImpl.java:255)
	at org.springframework.boot.web.embedded.undertow.UndertowServletWebServerFactory.createManager(UndertowServletWebServerFactory.java:329)
	at org.springframework.boot.web.embedded.undertow.UndertowServletWebServerFactory.getWebServer(UndertowServletWebServerFactory.java:298)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.createWebServer(ServletWebServerApplicationContext.java:183)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:161)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:578)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:730)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:432)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:308)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1302)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1291)
	at org.myspring.MainApplication.main(MainApplication.java:14)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:95)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65)

2022-12-13T16:04:38.235+08:00  INFO 127368 --- [main] org.joinfaces.ClasspathScanUtil          : Loading prepared scan result took 26.7039ms
2022-12-13T16:04:39.647+08:00  INFO 127368 --- [main] .w.PostConstructApplicationEventListener : Running on PrimeFaces 12.0.0
2022-12-13T16:04:39.649+08:00  INFO 127368 --- [main] .w.PostConstructApplicationEventListener : Running on PrimeFaces 12.0.0
2022-12-13T16:04:39.649+08:00  INFO 127368 --- [main] .a.PostConstructApplicationEventListener : Running on PrimeFaces Extensions 12.0.2
2022-12-13T16:04:39.649+08:00  INFO 127368 --- [main] .a.PostConstructApplicationEventListener : Running on PrimeFaces Extensions 12.0.2
2022-12-13T16:04:39.810+08:00  INFO 127368 --- [main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2022-12-13T16:04:40.152+08:00  INFO 127368 --- [main] com.zaxxer.hikari.pool.HikariPool        : HikariPool-1 - Added connection org.postgresql.jdbc.PgConnection@65d57e4e
2022-12-13T16:04:40.154+08:00  INFO 127368 --- [main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
2022-12-13T16:04:40.300+08:00  INFO 127368 --- [main] o.hibernate.jpa.internal.util.LogHelper  : HHH000204: Processing PersistenceUnitInfo [name: default]
2022-12-13T16:04:40.420+08:00  INFO 127368 --- [main] org.hibernate.Version                    : HHH000412: Hibernate ORM core version 6.1.5.Final
2022-12-13T16:04:40.748+08:00  WARN 127368 --- [main] org.hibernate.orm.deprecation            : HHH90000021: Encountered deprecated setting [javax.persistence.sharedCache.mode], use [jakarta.persistence.sharedCache.mode] instead
2022-12-13T16:04:40.941+08:00  INFO 127368 --- [main] SQL dialect                              : HHH000400: Using dialect: org.hibernate.dialect.PostgreSQLDialect
2022-12-13T16:04:41.620+08:00  INFO 127368 --- [main] o.h.e.t.j.p.i.JtaPlatformInitiator       : HHH000490: Using JtaPlatform implementation: [org.hibernate.engine.transaction.jta.platform.internal.NoJtaPlatform]
2022-12-13T16:04:41.626+08:00  INFO 127368 --- [main] j.LocalContainerEntityManagerFactoryBean : Initialized JPA EntityManagerFactory for persistence unit 'default'
2022-12-13T16:04:41.653+08:00  WARN 127368 --- [main] ocalVariableTableParameterNameDiscoverer : Using deprecated '-debug' fallback for parameter name resolution. Compile the affected code with '-parameters' instead or avoid its introspection: org.joinfaces.autoconfigure.i18n.MessageSourceBridgeAutoConfiguration
2022-12-13T16:04:42.008+08:00  INFO 127368 --- [main] io.undertow                              : starting server: Undertow - 2.3.0.Final
2022-12-13T16:04:42.038+08:00  INFO 127368 --- [main] org.xnio                                 : XNIO version 3.8.8.Final
2022-12-13T16:04:42.066+08:00  INFO 127368 --- [main] org.xnio.nio                             : XNIO NIO Implementation Version 3.8.8.Final
2022-12-13T16:04:42.228+08:00  INFO 127368 --- [main] org.jboss.threads                        : JBoss Threads version 3.5.0.Final
2022-12-13T16:04:42.306+08:00  INFO 127368 --- [main] o.s.b.w.e.undertow.UndertowWebServer     : Undertow started on port(s) 8080 (http)
2022-12-13T16:04:42.316+08:00  INFO 127368 --- [main] org.myspring.MainApplication             : Started MainApplication in 17.988 seconds (process running for 18.791)
```

**Comment by larsgrefer** [comment]
Can you provide an example project which I can test?

**Comment by dengbingaliyun** [comment]
I have sent examples to github@larsgrefer.de , please check.

**Comment by joensen** [comment]
I am having the same issue and would also like it resolved.

It only affects startup time and the log statements are at WARN level. The application still starts.
I have therefore, until this issue is resolved, disabled logging from weld.

logging.level.org.jboss.weld = ERROR

**Comment by larsgrefer** [comment]
Could you try angain with JoinFaces 5.0.1?

**Comment by dengbingaliyun** [comment]
I have tried with version 5.0.1 ,but the issue still remain the same.
The alternative way is to add  a line  in application.properties to disable error logging as [joensen](https://github.com/joensen) 
 did ，the code list below:
`logging.level.org.jboss.weld.Bootstrap=OFF`

**Comment by larsgrefer** [comment]
Seems like Jandex does not like Spring Boot repackaged jars.

**Comment by jomu78** [comment]
@larsgrefer : works for me with JoinFaces 5.0.2, see example app https://github.com/jomu78/tests/tree/main/joinfaces-adminfaces-jdk17

**Comment by dengbingaliyun** [comment]
The issue solved in version 5.0.2，thanks Lars. Love joinfaces all the time.
